site/app/controllers/ChatroomController.php

```diff
@@ -0,0 +1,234 @@
+use app\entities\chat\Message;
+use app\libraries\routers\AccessControl;
+use app\libraries\routers\Enabled;
+use app\views\ChatroomView;
+use Symfony\Component\Routing\Annotation\Route;
+use Doctrine\ORM\EntityManagerInterface;
+
+class ChatroomController extends AbstractController {
+
+    public function __construct(Core $core) {
+        parent::__construct($core);
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat", methods={"GET"})
+     */
+    public function showChatroomsPage(): WebResponse {
+        $repo = $this->core->getCourseEntityManager()->getRepository(Chatroom::class);
+        $user = $this->core->getUser();
+        $chatrooms = $repo->findBy([], ['id' => 'ASC']);
+        $active_chatrooms = $repo->findBy(['is_active' => true], ['id' => 'ASC']);
+
+        if ($user->accessAdmin()) {
+            return new WebResponse(
+                'Chatroom',
+                'showChatPageInstructor',
+                $chatrooms
+            );
+        }
+        else { // Student view
+            return new WebResponse(
+                'Chatroom',
+                'showChatPageStudent',
+                $active_chatrooms
+            );
+        }
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/newChatroom", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function addChatroom(): RedirectResponse {
+        $em = $this->core->getCourseEntityManager();
+
+        $chatroom = new Chatroom();
+        $chatroom->setTitle($_POST['title']);
+        $chatroom->setDescription($_POST['description']);
+        $chatroom->setHostId($this->core->getUser()->getId());
+        $chatroom->setHostName($this->core->getUser()->getDisplayFullName());
+
+        $em->persist($chatroom);
+        $em->flush();
+
+        $this->core->addSuccessMessage("Chatroom successfully added");
+        return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/{chatroom_id}", methods={"GET"})
+     * @param string $chatroom_id
+     * @return RedirectResponse|WebResponse
+     */
+     public function getChatroom(string $chatroom_id): WebResponse|RedirectResponse {
+         if (!is_numeric($chatroom_id)) {
+            $this->core->addErrorMessage("Invalid Chatroom ID");
+            return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+        }
+
+        $repo = $this->core->getCourseEntityManager()->getRepository(Chatroom::class);
+        $chatroom = $repo->find($chatroom_id);
+
+        return new WebResponse(
+            'Chatroom',
+            'showChatroom',
+            $chatroom,
+        );
+     }
+
+     /**
+     * @Route("/courses/{_semester}/{_course}/chat/{chatroom_id}/anonymous", methods={"GET"})
+     * @param string $chatroom_id
+     * @return RedirectResponse|WebResponse
+     */
+     public function getChatroomAnon(string $chatroom_id): WebResponse|RedirectResponse {
```

Comment:
What's the purpose of this route?  It appears to be essentially identical to the route without the `/anonymous`